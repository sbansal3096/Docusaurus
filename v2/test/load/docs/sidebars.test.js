import path from 'path';
import loadSidebars from '@lib/load/docs/sidebars';

describe('loadSidebars', () => {
  const fixtures = path.join(__dirname, '..', '__fixtures__');
  test('normal site with sidebars', () => {
    const env = {};
    const siteDir = path.join(fixtures, 'simple-site');
    const result = loadSidebars({siteDir, env});
    expect(result).toMatchSnapshot();
  });

  test('site without sidebars', () => {
    const env = {};
    const siteDir = path.join(fixtures, 'bad-site');
    const result = loadSidebars({siteDir, env});
    expect(result).toMatchSnapshot();
  });

  test('site with sidebars & versioned sidebars', () => {
    const env = {
      versioning: {
        enabled: true,
        versions: ['1.0.1', '1.0.0'],
      },
    };
    const siteDir = path.join(fixtures, 'versioned-site');
    const result = loadSidebars({siteDir, env});
    expect(result).toMatchSnapshot();
  });

  test('site with missing versioned sidebars', () => {
    const env = {
      versioning: {
        enabled: true,
        versions: ['2.0.0'],
      },
    };
    const siteDir = path.join(fixtures, 'versioned-site');
    expect(() => {
      loadSidebars({siteDir, env});
    }).toThrowErrorMatchingInlineSnapshot(
      `"Failed to load versioned_sidebars/version-2.0.0-sidebars.json. It does not exist."`,
    );
  });
});
