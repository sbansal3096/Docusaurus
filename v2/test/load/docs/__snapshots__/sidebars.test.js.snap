// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadSidebars normal site with sidebars 1`] = `
Object {
  "docs": Object {
    "Getting Started": Array [
      "installation",
    ],
    "Guides": Array [
      "blog",
    ],
  },
}
`;

exports[`loadSidebars site with sidebars & versioned sidebars 1`] = `
Object {
  "docs": Object {
    "Getting Started": Array [
      "installation",
    ],
    "Guides": Array [
      "blog",
    ],
  },
  "version-1.0.0-docs": Object {
    "Getting Started": Array [
      "version-1.0.0-installation",
    ],
    "Guides": Array [
      "version-1.0.0-blog",
    ],
  },
  "version-1.0.1-docs": Object {
    "Getting Started": Array [
      "version-1.0.1-installation",
    ],
    "Guides": Array [
      "version-1.0.1-blog",
    ],
  },
}
`;

exports[`loadSidebars site without sidebars 1`] = `Object {}`;
